doc/developers/performance.rst

```diff
@@ -46,31 +46,31 @@ Sometimes however an algorithm cannot be expressed efficiently in simple
 vectorized Numpy code. In this case, the recommended strategy is the
 following:
 
-  1. **Profile** the Python implementation to find the main bottleneck and
-     isolate it in a **dedicated module level function**. This function
-     will be reimplemented as a compiled extension module.
-
-  2. If there exists a well maintained BSD or MIT **C/C++** implementation
-     of the same algorithm that is not too big, you can write a
-     **Cython wrapper** for it and include a copy of the source code
-     of the library in the scikit-learn source tree: this strategy is
-     used for the classes :class:`svm.LinearSVC`, :class:`svm.SVC` and
-     :class:`linear_model.LogisticRegression` (wrappers for liblinear
-     and libsvm).
-
-  3. Otherwise, write an optimized version of your Python function using
-     **Cython** directly. This strategy is used
-     for the :class:`linear_model.ElasticNet` and
-     :class:`linear_model.SGDClassifier` classes for instance.
-
-  4. **Move the Python version of the function in the tests** and use
-     it to check that the results of the compiled extension are consistent
-     with the gold standard, easy to debug Python version.
-
-  5. Once the code is optimized (not simple bottleneck spottable by
-     profiling), check whether it is possible to have **coarse grained
-     parallelism** that is amenable to **multi-processing** by using the
-     ``joblib.Parallel`` class.
+1. **Profile** the Python implementation to find the main bottleneck and
+    isolate it in a **dedicated module level function**. This function
+    will be reimplemented as a compiled extension module.
+
+2. If there exists a well maintained BSD or MIT **C/C++** implementation
+    of the same algorithm that is not too big, you can write a
+    **Cython wrapper** for it and include a copy of the source code
+    of the library in the scikit-learn source tree: this strategy is
+    used for the classes :class:`svm.LinearSVC`, :class:`svm.SVC` and
+    :class:`linear_model.LogisticRegression` (wrappers for liblinear
+    and libsvm).
+
+3. Otherwise, write an optimized version of your Python function using
+    **Cython** directly. This strategy is used
+    for the :class:`linear_model.ElasticNet` and
+    :class:`linear_model.SGDClassifier` classes for instance.
+
+4. **Move the Python version of the function in the tests** and use
+    it to check that the results of the compiled extension are consistent
+    with the gold standard, easy to debug Python version.
+
+5. Once the code is optimized (not simple bottleneck spottable by
+    profiling), check whether it is possible to have **coarse grained
+    parallelism** that is amenable to **multi-processing** by using the
+    ``joblib.Parallel`` class.
```

Comment:
```suggestion
1. **Profile** the Python implementation to find the main bottleneck and
   isolate it in a **dedicated module level function**. This function
   will be reimplemented as a compiled extension module.

2. If there exists a well maintained BSD or MIT **C/C++** implementation
   of the same algorithm that is not too big, you can write a
   **Cython wrapper** for it and include a copy of the source code
   of the library in the scikit-learn source tree: this strategy is
   used for the classes :class:`svm.LinearSVC`, :class:`svm.SVC` and
   :class:`linear_model.LogisticRegression` (wrappers for liblinear
   and libsvm).

3. Otherwise, write an optimized version of your Python function using
   **Cython** directly. This strategy is used
   for the :class:`linear_model.ElasticNet` and
   :class:`linear_model.SGDClassifier` classes for instance.

4. **Move the Python version of the function in the tests** and use
   it to check that the results of the compiled extension are consistent
   with the gold standard, easy to debug Python version.

5. Once the code is optimized (not simple bottleneck spottable by
   profiling), check whether it is possible to have **coarse grained
   parallelism** that is amenable to **multi-processing** by using the
   ``joblib.Parallel`` class.
```